test/extended/etcd/helpers/helpers.go

```diff
@@ -180,6 +183,91 @@ func recoverClusterToInitialStateIfNeeded(ctx context.Context, t TestingT, machi
 	})
 }
 
+func DeleteSingleMachine(ctx context.Context, t TestingT, machineClient machinev1beta1client.MachineInterface) (string, error) {
+	waitPollInterval := 15 * time.Second
+	waitPollTimeout := 5 * time.Minute
+	t.Logf("Waiting up to %s to delete a machine", waitPollTimeout.String())
+
+	machineToDelete := ""
+	err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		machineList, err := machineClient.List(ctx, metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
```

Comment:
Why does is this listing repeated instead of identifying the machine-to-be-deleted once and then retrying the Delete until you get a successful call?

Reply:
That's to retry the List if it errors out due to a transient API error. Would we not want to retry in that case?

Reply:
> That's to retry the List if it errors out due to a transient API error. Would we not want to retry in that case?

The client retries automatically.  Failures at this level aren't expected and even in controller code aren't generally handled.